apis/go.mod

```diff
@@ -1,36 +1,38 @@
 module github.com/openstack-k8s-operators/infra-operator/apis
 
-go 1.19
+go 1.21
+
+toolchain go1.21.1
```

Comment:
OK, this is a min constraint and go will use whatever that is newer and built into the available go 1.21.x installation

Reply:
I guess this is automatically added by something. But for consistency, I'll add it to the root go.mod too.